packages/ibm-products/src/components/StatusIndicator/StatusIndicator.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright IBM Corp. 2024, 2024
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import { getDevtoolsProps } from '../../global/js/utils/devtools';
+import { pkg } from '../../settings';
+
+// Carbon and package components we use.
+import { Button } from '@carbon/react';
+import { Restart } from '@carbon/react/icons';
+
+// The block part of our conventional BEM class names (blockClass__E--M).
+const blockClass = `${pkg.prefix}--status-indicator`;
+const componentName = 'StatusIndicator';
+
+const defaults = {
+  retryLabel: 'Retry',
+  showRetry: false,
+};
+
+/**
+ * A list of icon/description pairs used to show multiple states of progress.
+ */
+export let StatusIndicator = React.forwardRef(
+  (
+    {
+      children,
+      className,
+      onRetry,
+      retryLabel = defaults.retryLabel,
+      showRetry = defaults.showRetry,
+      title,
+      ...rest
+    },
+    ref
+  ) => {
+    return (
+      <div
+        {...rest}
+        className={cx(blockClass, className)}
+        ref={ref}
+        {...getDevtoolsProps(componentName)}
+      >
+        {title && <h1 className={`${blockClass}__title`}>{title}</h1>}
+        <ul className={`${blockClass}__list`}>{children}</ul>
+        {showRetry && onRetry && retryLabel && (
+          <Button
+            className={`${blockClass}__button`}
+            kind="ghost"
+            onClick={onRetry}
+            onKeyUp={onRetry}
+            renderIcon={() => <Restart size={16} />}
+          >
+            {retryLabel}
+          </Button>
+        )}
+      </div>
+    );
+  }
+);
+
+// Return a placeholder if not released and not enabled by feature flag
+StatusIndicator = pkg.checkComponentEnabled(StatusIndicator, componentName);
+
+// The display name of the component, used by React. Note that displayName
+// is used in preference to relying on function.name.
+StatusIndicator.displayName = componentName;
+
+// The types and DocGen commentary for the component props,
+// in alphabetical order (for consistency).
+// See https://www.npmjs.com/package/prop-types#usage.
+StatusIndicator.propTypes = {
+  /**
+   * Provide the contents of the StatusIndicator.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Provide an optional class to be applied to the containing node.
+   */
+  className: PropTypes.string,
+  /**
+   * Optional callback function for the retry button.
+   */
+  onRetry: PropTypes.func,
+  /**
+   * The text for the retry button.
+   */
+  retryLabel: PropTypes.string,
+  /**
+   * Set to `true` to show the retry button.
+   *
+   * `retryLabel` and `onRetry` must also be defined.
+   */
+  showRetry: PropTypes.bool,
```

Comment:
What do you think about only including the `onRetry` prop? If `onRetry` is passed in then you know you want to show the retry button. I don't see any other logic in terms of showing the retry button, could be an easy way to reduce the total number of props.

Reply:
Originally, I was thinking the same thing. And then...

Normally, buttons like this are always visible, it's just their state that changes based on conditions: disabled, busy, enabled, etc.

For the status indicator, the expected behavior is that this button is rendered only if any status returns an error. This is atypical.

More intuitive and self-documenting. "Show the button if there's an error".
```
hasError: useState(boolean)

onRetry: () => { do something }
retryLabel: "Retry"
showRetry: hasError
```

Less intuitive. "The button has a function if there's an error" does not imply that the button will be rendered or not.
```
hasError: useState(boolean)

onRetry: hasError ? undefined : () => { do something }
retryLabel: "Retry"
```

Reply:
> could be an easy way to reduce the total number of props

This is a very simple component, and there are only four props, excluding `children` and `className`.

Reply:
Thanks for the detailed explanation here, very helpful especially for other components moving forward 😄